pkg/gencred/README.md

```diff
@@ -0,0 +1,100 @@
+# Gencred
+
+## Description
+
+`gencred` is a simple tool used to generate cluster auth credentials (w/ [**cluster-admin**](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#user-facing-roles) permissions) for authenticating to a Kubernetes cluster.
+> **NOTE:** since `gencred` creates credentials with `cluster-admin` level access, the kube context used **must** also be bound to the `cluster-admin` role.
```

Comment:
Can we check that this is the case?

Check out `kubectl auth can-i`

Reply:
Should we do this here, in `mkbuild-cluster`, or both places?

Reply:
I think this is valuable, but I would like to tackle it as part of https://github.com/kubernetes/test-infra/issues/11011 and apply it directly to `mkbuild-cluster` or expose as a separate function.